Sources/BraintreeThreeDSecure/BTThreeDSecureV2Provider.swift

```diff
@@ -0,0 +1,160 @@
+import Foundation
+import CardinalMobile
+
+#if canImport(BraintreeCore)
+import BraintreeCore
+#endif
+
+// TODO: Can be internal and maybe a struct once BTThreeDSecureRequest is in Swift
+@objcMembers public class BTThreeDSecureV2Provider: NSObject {
+
+    // MARK: - Internal Properties
+
+    let cardinalSession: CardinalSession
+    let apiClient: BTAPIClient
+
+    var lookupResult: BTThreeDSecureResult? = nil
+
+    static var completionHandler: (BTThreeDSecureResult?, Error?) -> Void = { _, _ in }
```

Comment:
I've never seen this before. Makes this completion handler optional, right?

Reply:
Good question - yeah we use this pattern a few times in the SDK (BTVenmoClient.appSwitchCompletion is another example). This initializes the completion with a default implementation, though it doesn't do anything until we use it. It doesn't need to be static though I am now realizing so going to update that.